indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -1163,13 +1172,15 @@ static class TaskEntry
   {
     private TaskInfo taskInfo;
 
+    private final DateTime createdTime;
     private DateTime lastUpdatedTime;
     private ListenableFuture<TaskStatus> future = null;
     private boolean isComplete = false;
 
-    TaskEntry(TaskInfo taskInfo)
+    TaskEntry(TaskInfo taskInfo, DateTime createdTime)
     {
       this.taskInfo = taskInfo;
+      this.createdTime = createdTime;
       this.lastUpdatedTime = DateTimes.nowUtc();
```

Comment:
Q: Should the `lastUpdatedTime` also be assigned the `createdTime` in the constructor?

I guess the difference between `createdTime` and `DateTimes.nowUtc()` is that the former is the time when the entry was _supposed_ to be updated/created and the latter is the time when it was _actually_ created.

The `lastUpdatedTime` is used to determine which of two competing updates should win in a race condition. Assigning it to `createdTime` in the constructor might ensure that the operation which actually came later is honored.

What do you think, @jtuglu1 ?